### PR TITLE
Initial commit with reviewed and edited articles

### PR DESCRIPTION
**Pull request by mauricesharp-okta**
## Preview link
7/21/ 4:35PM EDT
https://64bae6bb2f20a10e899de976--reverent-murdock-829d24.netlify.app/docs/guides/terraform-landing-page/main/

## Description:
- First release of Terraform Docs
- Integrate terrform into the docs hiearchy
- Add to Guides overview page
- Section and articles for Terraform including a landing page for the top-level left-hand menu link "Automate managing with Terraform"

### Resolves:

* [OKTA-619533](https://oktainc.atlassian.net/browse/OKTA-619533)

### CODE REVIEW

File: packages/@okta/vuepress-theme-prose/const/navbar.const.js [hunk 1]
```diff
@@ -707,6 +707,44 @@ export const guides = [
...
+          {
+            title: "Manage authentication services",
+            guideName: "terraform-manage-external-authenticators",
+          },
+          {
+            title: "Manager end-user experience",
+            guideName: "terraform-manage-end-user-experience",
+          },
+          {
+            title: "Design secure Terraform access",
```
Comment (brentschaus-okta) [inline packages/@okta/vuepress-theme-prose/const/navbar.const.js:739]: How come the title here and the h1 in the guide are different?

File: packages/@okta/vuepress-site/docs/guides/terraform-manage-user-access/main/index.md [hunk 2]
```diff
@@ -0,0 +1,442 @@
...
+* Configure a passwordless sign-in flow using the Okta Terraform Provider
+* Configure multifactor authentication (MFA) with Okta Verify using the Okta Terraform Provider
+* Enforce priority order for policies and rules
+
+**What you need**
+
+* Familiarity with the Terraform terms: configuration, resources, state, and commands. See the [Terraform overview](/docs/guides/terraform-overview).
+* [Okta Developer Edition organization](https://developer.okta.com/signup) or an Okta Identity Engine organization
+* A Terraform configuration that can access your Okta org. See [Enable Terraform access to your Okta org](/docs/guides/terraform-enable-org-access).
+
```
Comment (brentschaus-okta) [inline packages/@okta/vuepress-site/docs/guides/terraform-manage-user-access/main/index.md:22]:
```suggestion
---
```

File: packages/@okta/vuepress-site/docs/guides/terraform-manage-groups/main/index.md [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+* Configure automatic group assignment for users in your org.
+* Use groups to manage user access to Okta applications.
+* Use groups to manage user sign-in flows.
+
+**What you need**
+
+* Familiarity with the Terraform terms: configuration, resources, state, and commands. See [Terraform Overview](/docs/guides/terraform-overview).
+* An [Okta Developer Edition organization](https://developer.okta.com/signup) or an Okta Identity Engine organization
+* A Terraform configuration that can access your Okta org. See [Enable Terraform access to your Okta org](/docs/guides/terraform-enable-org-access).
+
```
Comment (brentschaus-okta) [inline packages/@okta/vuepress-site/docs/guides/terraform-manage-groups/main/index.md:22]:
```suggestion
---
```

File: packages/@okta/vuepress-site/docs/guides/terraform-manage-external-authenticators/main/index.md [hunk 4]
```diff
@@ -0,0 +1,328 @@
...
+* Configure automatic creation of an Okta account the first time a user signs in using an external account.
+
+**What you need**
+
+* Familiarity with the Terraform terms: configuration, resources, state, and commands. See the [Terraform overview](/docs/guides/terraform-overview).
+* An [Okta Developer Edition organization](https://developer.okta.com/signup) or an Okta Identity Engine organization.
+* A Terraform configuration that can access your Okta org. See [Enable Terraform access to your Okta org](/docs/guides/terraform-enable-org-access).
+* An account with [Google](https://accounts.google.com/signup).
+* Access to your [Google Admin Console](admin.google.com) for SAML IdP creation.
+
```
Comment (brentschaus-okta) [inline packages/@okta/vuepress-site/docs/guides/terraform-manage-external-authenticators/main/index.md:24]:
```suggestion
---
```

File: packages/@okta/vuepress-site/docs/guides/terraform-design-rate-limits/main/index.md [hunk 5]
```diff
@@ -0,0 +1,147 @@
...
+Optimize your configuration to reduce the number of API calls that Terraform makes, and set custom rate limits to stop Terraform before you reach your org’s rate limits.
+
+---
+
+**Learning outcomes**
+
+* Optimize the number of API calls in your Terraform configuration.
+* Set a maximum rate for Terraform API usage.
+* Set custom rate limits in the Okta service application used by Terraform.
+
```
Comment (brentschaus-okta) [inline packages/@okta/vuepress-site/docs/guides/terraform-design-rate-limits/main/index.md:16]:
```suggestion
---
```

**Review by brentschaus-okta — APPROVED** [review]
LGTM

just a few nits and a question

File: packages/@okta/vuepress-theme-prose/const/navbar.const.js [hunk 6]
```diff
@@ -707,6 +707,44 @@ export const guides = [
           },
         ],
       },
+      {
+        title: "Automate managing with Terraform",
```
[the remark this answers is not in the document]
Comment (susanharper-okta) [reply packages/@okta/vuepress-theme-prose/const/navbar.const.js:711]: @mauricesharp-okta what happened to the original title of: Automate managing your Okta org with Terraform

File: packages/@okta/vuepress-theme-prose/const/navbar.const.js (hunk 1 above)
Comment (mauricesharp-okta) [reply packages/@okta/vuepress-theme-prose/const/navbar.const.js:739]: Good catch! The next commit fixes this (I hope).

File: packages/@okta/vuepress-theme-prose/const/navbar.const.js [hunk 7]
```diff
@@ -707,6 +707,44 @@ export const guides = [
...
+          {
+            title: "Manage groups",
+            guideName: "terraform-manage-groups",
+          },
+          {
+            title: "Manage authentication services",
+            guideName: "terraform-manage-external-authenticators",
+          },
+          {
+            title: "Manager end-user experience",
```
Comment (brianduffield-okta) [inline packages/@okta/vuepress-theme-prose/const/navbar.const.js]:
```suggestion
            title: "Manage end-user experience",
```

File: packages/@okta/vuepress-site/docs/guides/terraform-design-access-security/main/index.md [hunk 8]
```diff
@@ -0,0 +1,194 @@
...
+
+To implement OAuth 2.0 authorization for Terraform, see [Enable Terraform access for your Okta org](/docs/guides/terraform-enable-org-access).
+
+### Configure an app for authorization
+
+Use an Okta API service app for machine-to-machine communication with Terraform. A service app authorizes Terraform using the [Client Credentials flow](/docs/guides/implement-grant-type/clientcreds/main/).
+
+With this flow, Terraform uses credentials to request access to your org. The service app verifies the credentials and authorizes Terraform access to Okta.
+
+The credentials used for this flow are a public/private key pair. Okta stores the public key in the service app, and Terraform uses the private key in the configuration. You can [generate the key pair] in the Okta service app or with your own secure internal methods. See [Create access credentials](/docs/guides/terraform-enable-org-access#create-access-credentials) in [Enable Terraform access for your Okta org](/docs/guides/terraform-enable-org-access).
```
Comment (brianduffield-okta) [inline packages/@okta/vuepress-site/docs/guides/terraform-design-access-security/main/index.md:43]: Looks like `[generate the key pair]` is missing the url

**Review by brianduffield-okta — COMMENTED** [review]
Found a few nits. Overall, a nice set of docs.

File: packages/@okta/vuepress-site/docs/guides/terraform-manage-groups/main/index.md [hunk 9]
```diff
@@ -0,0 +1,175 @@
+---
+title: Manage groups of users with Terraform
```
[the remark this answers is not in the document]
Comment (mauricesharp-okta) [reply packages/@okta/vuepress-site/docs/guides/terraform-manage-groups/main/index.md:2]: That one had a back and forth that resulted in the current name... it was once just "groups" but it was deemed too Okta Jargon specific.

File: packages/@okta/vuepress-site/docs/guides/terraform-manage-groups/index.md [hunk 10]
```diff
@@ -0,0 +1,9 @@
+---
+title: Manage groups of users with Terraform
```
[the remark this answers is not in the document]
Comment (mauricesharp-okta) [reply packages/@okta/vuepress-site/docs/guides/terraform-manage-groups/index.md:2]: Same comment.

File: packages/@okta/vuepress-theme-prose/const/navbar.const.js (hunk 7 above)
Comment (mauricesharp-okta) [reply packages/@okta/vuepress-theme-prose/const/navbar.const.js]: Thanks... and I caught that one while fixing another bug from Brent. It's just not committed yet.

File: packages/@okta/vuepress-site/docs/guides/terraform-design-access-security/main/index.md (hunk 8 above)
Comment (mauricesharp-okta) [reply packages/@okta/vuepress-site/docs/guides/terraform-design-access-security/main/index.md:43]: Thanks... I put in a link to an article on implementing OAuth for a service app... I'm not 100% sure of the link as I didn't write this.

**Pull request merged**